src/mtype.d

```diff
@@ -8089,7 +8089,12 @@ extern (C++) final class TypeStruct : Type
     {
         //printf("TypeStruct::semantic('%s')\n", sym.toChars());
         if (deco)
+        {
+            if (sc.cppmangle != CPPMANGLE.def && this.cppmangle == CPPMANGLE.def)
+                this.cppmangle = sc.cppmangle;
+            assert(sc.cppmangle == CPPMANGLE.def || this.cppmangle == sc.cppmangle);
```

Comment:
This looks like some redundant logic. How about the equivalent with fewer operations:
```
if (sc.cppmangle != CPPMANGLE.def)
{
    if (this.cppmangle == CPPMANGLE.def)
        this.cppmangle = sc.cppmangle;
    else
         assert(this.cppmangle == sc.cppmangle);
}
```